core/src/main/java/com/google/zxing/oned/OneDimensionalCodeWriter.java

```diff
@@ -144,5 +144,15 @@ public int getDefaultMargin() {
    * @return a {@code boolean[]} of horizontal pixels (false = white, true = black)
    */
   public abstract boolean[] encode(String contents);
+
+  /**
+   * Can be overwritten if the encode requires to read the hints map. Otherwise it defaults to {@code encode}.
+   * @param contents barcode contents to encode
+   * @param hints encoding hints
+   * @return a {@code boolean[]} of horizontal pixels (false = white, true = black)
+   */
+  protected boolean[] encodeWithHints(String contents, Map<EncodeHintType,?> hints) {
```

Comment:
This is OK; you're welcome to move these two encode() methods up with the other encode() methods too

Reply:
Sorry I don't quite understand. You mean just to move the lines up to e.g. line 76 to where it's used? Or do your mean up the inheritance tree into the Writer?

Reply:
Just moving the definitions. I don't know why I put this method at the end when the other encode methods are at the top. All of them including the new one could go at the top together.

Reply:
Ok, makes sense. I also changed the name from encodeWithHints to just encode. I mainly write C, so I often forget overloading exists. ;)